Assets/Scripts/Persona/utils/ViewVisibilityOnScreen.cs

```diff
@@ -5,22 +5,19 @@ namespace IO.Persona.MobileAds.Unity
 {
     public class ViewVisibilityOnScreen
     {
-        public static float GetVisiblePercentage(RectTransform rectTransform, Canvas canvas)
+        public static float GetVisiblePercentage(RectTransform targetRectTransform, RectTransform canvasRectTransform)
         {
-            // Get the Canvas RectTransform (assuming the Canvas is the parent of the image)
-            RectTransform canvasRectTransform = canvas.GetComponent<RectTransform>();
-
             // Convert RectTransforms to screen space Rects
-            Rect imageRectangle = GetScreenSpaceRect(rectTransform);
+            Rect targetRectangle = GetScreenSpaceRect(targetRectTransform);
             Rect canvasRectangle = GetScreenSpaceRect(canvasRectTransform);
 
             // Calculate intersecting area
-            Rect intersection = CalculateIntersection(imageRectangle, canvasRectangle);
+            Rect intersectionRectangle = GetIntersectionRect(targetRectangle, canvasRectangle);
 
             // Calculate and display intersecting area's size
-            float visibleImageArea = imageRectangle.width * imageRectangle.height;
-            float intersectingArea = intersection.width * intersection.height;
-            float percentageVisible = (intersectingArea / visibleImageArea) * 100f;
+            float targetRectangleArea = targetRectangle.width * targetRectangle.height;
+            float intersectionRectangleArea = intersectionRectangle.width * intersectionRectangle.height;
+            float percentageVisible = (intersectionRectangleArea / targetRectangleArea) * 100f;
```

Comment:
We should add a null | 0 check for targetRectangleArea's value

Reply:
yeah ok adding this